packages/block-editor/src/components/inserter/block-patterns-tab.js

```diff
@@ -138,35 +84,204 @@ function BlockPatternsTabs( {
 		return categories;
 	}, [ allPatterns, allCategories ] );
 
-	const patternCategory = selectedCategory
-		? selectedCategory
-		: populatedCategories[ 0 ];
+	return populatedCategories;
+}
+
+export function BlockPatternsCategoryDialog( {
+	rootClientId,
+	onInsert,
+	category,
+	onClose,
+} ) {
+	const container = useRef();
+	const [ ref, props ] = useDialog( {
+		onClose,
+	} );
+
+	useEffect( () => {
+		container?.current?.focus();
```

Comment:
Is the `container` ref needed? What's the difference between this and `ref?.current?.focus()`?

Reply:
Ref is returned from useDialog, we don't control its shape here so it could be an object with "current" or it could be a callback (functions are valid refs). So Adding our own self controlled ref is safer.